src/pages/Group/Group.tsx

```diff
@@ -68,6 +70,36 @@ export default function Group() {
     () => api.get<Group>("/group/" + groupID).then((res) => res.data),
   );
 
+  const [startDate, setStartDate] = useState(new Date());
+  const [endDate, setEndDate] = useState(new Date());
+  const [radioValue, setRadioValue] = useState(1);
+
+  // Changes selected date range between current week and current month when radioValue variable changes.
+  useEffect(() => {
+    setStartDate(() => {
+      if (radioValue === 1) {
+        return new Date(
+          startDate.getFullYear(),
+          startDate.getMonth(),
+          startDate.getDate() - startDate.getDay(),
+        );
+      } else {
+        return new Date(endDate.getFullYear(), endDate.getMonth(), 1);
+      }
+    });
+    setEndDate(() => {
+      if (radioValue === 1) {
+        return new Date(
+          startDate.getFullYear(),
+          startDate.getMonth(),
+          startDate.getDate() - startDate.getDay() + 6,
+        );
+      } else {
+        return new Date(endDate.getFullYear(), endDate.getMonth() + 1, 0);
+      }
+    });
+  }, [radioValue]);
+
```

Comment:
I thought we had an abstraction for this already? is there a reason you have this same kind of code here?